sparkly/catalog.py

```diff
@@ -38,25 +38,52 @@ def drop_table(self, table_name, checkfirst=True):
             table_name (str): A table name.
             checkfirst (bool): Only issue DROPs for tables that are presented in the database.
         """
+        db_name = get_db_name(table_name)
+        if checkfirst and not self.has_database(db_name):
+            return
+
         drop_statement = 'DROP TABLE IF EXISTS' if checkfirst else 'DROP TABLE'
-        return self._spark.sql('{} `{}`'.format(drop_statement, table_name))
+        return self._spark.sql(
+            '{} {}'.format(drop_statement, table_name)
+        )
 
-    def has_table(self, table_name, db_name=None):
+    def has_table(self, table_name):
         """Check if table is available in the metastore.
 
         Args:
             table_name (str): A table name.
-            db_name (str): A database name.
 
         Returns:
             bool
         """
+        db_name = get_db_name(table_name)
```

Comment:
as a reminder you can do something like:
```
db, _ = get_db_and_table(table_name)
_, table = get_db_and_table(table_name)
```
But I don't have strong objections against your solution.